pkg/smartcontract/manifest/manifest.go

```diff
@@ -118,7 +118,19 @@ func (m *Manifest) IsValid(hash util.Uint160) error {
 			return errors.New("duplicate trusted contracts")
 		}
 	}
-	return Permissions(m.Permissions).AreValid()
+	err = Permissions(m.Permissions).AreValid()
+	if err != nil {
+		return err
+	}
+	si, err := m.ToStackItem()
+	if err != nil {
+		return fmt.Errorf("failed to check manifest serialisation: %w", err)
+	}
+	_, err = stackitem.Serialize(si)
```

Comment:
Which means we're serializing it twice. And btw, we're saving a full contract state that has more elements than just a manifest, maybe that's sufficient for the purpose (we have the same limit there anyway)? Not that serializing twice is that bad here, `IsValid()` is not on a hot path.

Reply:
> Which means we're serializing it twice.

You're right, but I thought that contracts deploy/update are rear so we may afford this. From the other hand, we may perform a tiny optimisation and try to serialize manifest, store serialized bytes in cache and then reuse this serialised array instead of reserializing, I'll try it.

> we have the same limit there anyway

Yes, we have exactly the same limit (it's inside `putConvertibleToDAO` and it is set implisitly), but maybe it's better to fail exatctly when C# node fails (just in case, although currently we don't have any significant code between manifest validness check and putting contract state to storage). And it's nice to have this check in `IsValid` because then compiler can use it and warn the user if manifest is too large.

Reply:
Although you know, let's remove this check. Transaction will fail anyway on large `state.Contract`, so it's OK to fail it a couple of lines later than C# node.

Reply:
See the latest commit.